packages/stream-tile/src/tile-document.ts

```diff
@@ -78,6 +93,27 @@ function headerFromMetadata(metadata?: TileMetadataArgs | StreamMetadata): Commi
     tags: metadata?.tags,
   }
 
+  // Handle properties that can only be set on the genesis commit.
+  if (genesis) {
+    if (!metadata?.deterministic) {
+      header.unique = uint8arrays.toString(randomBytes(12), 'base64')
+    }
+    if (metadata?.forbidControllerChange) {
+      header.forbidControllerChange = true
+    }
+  } else {
+    // These throws aren't strictly necessary as we can just leave these fields out of the header
+    // object we return, but throwing here gives more useful feedback to users.
+
+    if (metadata?.deterministic !== undefined || (metadata as any)?.unique !== undefined) {
+      throw new Error("Cannot change 'deterministic' or 'unique' properties on existing Streams")
```

Comment:
@stbrody I'm new to Ceramic, working through the basics, trying to do the following 

`await idx.set("basicProfile", { name: "testing " });` 

but am getting the above error when making this call. Have attempted to understand what could be causing this issue but so far no luck. As a new Ceramic user this error wasn't clear in what it means by deterministic and unique, perhaps that means something more specific to an experienced user?

Reply:
My issue was resolved by updating to `@ceramicstudio/idx` version `0.12.2`

https://github.com/ceramicnetwork/js-ceramic/issues/1714